lib/prometheus/client/metric.rb

```diff
@@ -29,16 +29,17 @@ def initialize(name,
         @docstring = docstring
         @preset_labels = stringify_values(preset_labels)
 
+        @all_labels_preset = false
+        if preset_labels.keys.length == labels.length
+          @validator.validate_labelset!(preset_labels)
+          @all_labels_preset = true
+        end
```

Comment:
Moved this block up since it makes sense to have it closer to the other `@label` variable definitions